### ISSUE TITLE
GG Max - Exp - Explanation of Health now ?

### ISSUE THREAD

**Issue by Tum1370** [TheGameCreators/GameGuruRepo#5288]
@LeeBamberTGC As the title says, Could you please explain how the Health works now in Max.

It use to work like this.

Starter Marker Health = Starting HP
MyHealthInitial = Maxmium HP (Unless skill points increase this)

But now this does not seem to be the case, Something must have changed again ?

I have my Starter Marker set at 100     (Which use to mean my character started with 100 HP)
I have my "MyHealthInitial" set at 100       (Which use to mean the maximum HP my character could eve have was 100)

But now, when i collect a health potion (Which uses the "Health.lua" , if i use the potion (which is set to give 25 HP), even if my HP is at 100, the potion will still add HP, and take my HP to 125.
This is now ignoring the "MyHealthInitial" stat which i have set at 100    (The suppose to be max HP)

Also i notice that the "Health.lua" script, affect the user global "MyGlobal"   ?????? What is "MyGlobal"  ??????

Is this a bug or working differently now ?

**Comment by Necrym59** [comment]
The maximum limite has now been fixed not to exceed the start maximum health level
The "MyGlobal" option is if you wish to increment another user global apart from the health at teh same time.

**Comment by Tum1370** [comment]
Hi, I don’t understand how you can set Maximum Health. 
in the HUD editor there is a Maximum Health on the left hand panel, but this seems to have no way to actually set Maximum Health. 

The health system currently seems no clear. 

MyHealthInitial. Means what actually?
Starter Marker Health, Means what etc ?

They all seem to be named incorrectly.

Surely it’d be much better having global like. 

MyHealthCurrent
MyHealthMaximum
MyHealthStart
MyHealthRegen

and so on. 

Currently it is not very user friendly.

**Comment by Necrym59** [comment]
OK which script are you using for the pickup -  health as you first stated or potion or somthing else? Standard install or workshop versions.  Nothing wrong with the health system as such, so lets work any problems without the carry on, it is exp software after all and there is always changes happening under the covers, sometimes things im not even aware of.

**Comment by Necrym59** [comment]
No it wont go above the 100 or whatever the startmarker amount set at ,  ive tested it  -update the workshop items and verify

**Comment by Tum1370** [comment]
It is updated. It still goes above 100. 

Whether this is Max doing it’s wonders and remembering old stuff I don’t know. 

But I pick up the item, my health goes up to 125. 

Max has some very weird bugs remembering old stuff.

**Comment by Necrym59** [comment]
It does work, i suggest your probably not using the latest one (health_v18) or its conflicting with somthing else try overwriting the steam installation version with the workshop version that should fix it. 
Ignore "MyHealthInitial" this was a probably a transition variable at the time. Forget all the malarky about the variable names they sometimes get modified according to updates and changes in code, the main ones are
"Health Remaining" and "Maximum Health" - The same two that are on display in the health panel.
"Maximum Health" is the health level that is set in the player panel marked "Health".
If by chance this does not work on a status bar then just report that fact and keep the report simple and concise and it will get looked at.

**Comment by davetgc** [comment]
I've tested this,. and I can't get the health to work with the status bar. Nothing changes when set.

**Comment by Tum1370** [comment]
@Necrym59 @LeeBamberTGC 
I checked my versions of health.lua
In my main GameGuruMax folder it had version 17.
In my mydocuments folder, it had version 18

So after copying over  the version from mydocuments to the main folder, I retested and it was now working as Necrym says it does. My health did not go above the 100 i had set.

Is Max suppose to automatically update the main scripts with the latest versions when we update workshop ? Because this didnt seem to happen here ?
